### PR TITLE
73 plugin application broken if project version is different than the language server library version

### LINKED ISSUE #73: Plugin application broken if project version is different than the language server library version
Lines like:

- https://github.com/Strumenta/kolasu-languageserver-library/blob/8d8e85dccc4869ac486cd186caee9afdd2b5a3a1/plugin/src/main/kotlin/com/strumenta/kolasu/languageserver/plugin/LanguageServerPlugin.kt#L31
- https://github.com/Strumenta/kolasu-languageserver-library/blob/8d8e85dccc4869ac486cd186caee9afdd2b5a3a1/plugin/src/main/kotlin/com/strumenta/kolasu/languageserver/plugin/LanguageServerPlugin.kt#L38

causes dependency resolution to fail as the `${project.version}` value is obtained from the properties of the project applying the plugin, not from those of the plugin itself. We should find another way to keep versions aligned among the various submodules - e.g. version catalogs. The same thing could be useful for aligning the version of Kolasu or other common dependencies among the modules.

### PR DESCRIPTION
**Pull request by loradd**
Fixes #73

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1 +1,6 @@
-version=1.0.4-SNAPSHOT
\ No newline at end of file
+version=1.0.4-SNAPSHOT
+kotlinVersion=1.8.22
+kolasuVersion=1.5.45
+lsp4jVersion=0.21.1
+luceneVersion=9.8.0
+junitVersion=5.7.1
```
Comment (martin-azpillaga) [inline gradle.properties:6]: I think centralizing all the versions makes sense.
Can we take the chance to centralize the versions of plugins such as ktlint or kotlin jvm as well?

File: library/build.gradle.kts [hunk 2]
```diff
@@ -78,3 +83,7 @@ publishing {
 signing {
     sign(publishing.publications.getByName("language-server-library"))
 }
+
+ktlint {
+    version = "1.2.1"
+}
```
Comment (martin-azpillaga) [inline library/build.gradle.kts:89]: Is this block necessary?
Do we intend to hardcode future versions of ktlint like this?
How about extracting it with the rest of versions too?

File: testing/build.gradle.kts [hunk 3]
```diff
@@ -78,3 +82,7 @@ publishing {
 signing {
     sign(publishing.publications.getByName("language-server-testing"))
 }
+
+ktlint {
+    version = "1.2.1"
+}
```
Comment (martin-azpillaga) [inline testing/build.gradle.kts:88]: Another option would be to extract these to the root gradle project

**Review by martin-azpillaga — APPROVED** [review]
Thanks for the contributions!

Regarding formatting changes, I am curious why these were not present before.
I hope that the formatting is standardized now and won't cause big changes in the future.

File: gradle.properties (hunk 1 above)
Comment (loradd) [reply gradle.properties:6]: Done, introduced version catalogs

File: library/build.gradle.kts (hunk 2 above)
Comment (loradd) [reply library/build.gradle.kts]: Nope, removed - the only block that remains required is the one ignoring BuildConfig stuff (they generate two-space indented code making ktlint upset)

File: testing/build.gradle.kts (hunk 3 above)
Comment (loradd) [reply testing/build.gradle.kts]: This would require making the root project a kotlin project, we might think about it in the future.

**Pull request merged**